src/components/common/ButtonWithIcon.vue

```diff
@@ -0,0 +1,47 @@
+<template>
+  <BaseButton :type="type" :disabled="disabled" :size="size">
+    <fa class="text-blue-400 mr-3" size="2x" :icon="icon"/>
+    <slot/>
+  </BaseButton>
+</template>
+
+<script>
+import BaseButton from "../../components/common/BaseButton";
+import {faStar} from "@fortawesome/free-solid-svg-icons";
+
+export default {
+  name: "ButtonWithIcon",
+  components: {BaseButton},
+  props: {
+    icon: {
```

Comment:
I wouldnt do this via props. Try using more of vues built in [Slot](https://vuejs.org/v2/guide/components-slots.html) system. 

This is what I used for the text `<slot />`

However you can use multiple ones for differnet components. 

```js
# ButtonWithIcon.vue
<slot name="text"/>
<slot name="icon"/>
```

```js
# ButtonWithIcon.stories.js
<template #text>TEXT GOES HERE</template>
<template #icon>ICON GOES HERE</template>
```